guacamole/src/main/webapp/app/client/controllers/clientController.js

```diff
@@ -219,6 +220,26 @@ angular.module('client').controller('clientController', ['$scope', '$routeParams
         remaining: 15
     };
 
+    /**
+     * Catch window or tab closing (ctrl-w) and prompt the user if there is an active connection
+     *
+     * @param {Event} e
+     * @returns {undefined}
+     */
+    var windowCloseListener = function onBeforeUnload(e) {
+        var managedClient = $scope.client;
+        if (managedClient) {
+
+            // Get current connection state
+            var connectionState = managedClient.clientState.connectionState;
+
+            // If connected, prompt to close
+            if (connectionState === ManagedClientState.ConnectionState.CONNECTED)
+                e.preventDefault();
+                e.returnValue = '';
```

Comment:
It's possible I'm just missing something here about how JS behaves, but does this actually prompt the user? It looks like all it does is check if the connection is active, block the default action, and clear the return value of the event? Does that somehow generate a browser prompt? Or does this just silently block the attempt to close the window?

Reply:
This event has special properties and causing the browser to generate a prompt is exactly what it does.  It used to be possible to provide a custom message via the returnValue, but that has been deprecated in pretty much all browsers for security reasons.

Reply:
Got it, thanks for clarifying!